src/controllers/projectCtrl.js

```diff
@@ -37,41 +37,27 @@ exports.deleteProject = (req, res) => {
 };
 
 exports.updateProjectInfos = (req, res) => {
-  //check if project exist
-  Project.findById(req.params.projectId)
+  Project.findByIdAndUpdate(req.params.projectId, req.body, { new: true })
     .then((project) => {
-      if (project) {
-        project = {
-          ...project,
-          ...req.body,
-        };
-        project
-          .save()
-          .then((newProject) =>
-            res.status(200).json({
-              message: 'Project infos updated!',
-              newProject,
-            })
-          )
-          .catch((error) => res.status(500).json(error));
-      } else {
-        res.status(404).json({
-          message: 'Project not found!',
-        });
-      }
+      if (!project)
+        res
+          .status(400)
+          .json({ message: 'No item updated: project not found!' });
+      else res.status(200).json({ message: 'Project updated!', data: project });
     })
     .catch((error) =>
-      res.status(401).json({ message: 'Invalid Request!', error })
+      res.status(400).json({ message: 'Invalid Request!', error })
     );
 };
 
 exports.addUserToProject = (req, res) => {
-  //check if project exist
+  // check if project exist
   Project.findById(req.params.projectId)
+    // eslint-disable-next-line consistent-return
     .then((project) => {
       if (project) {
         const user = project.users.find(
-          (user) => user.toString() === req.body.userId
+          (u) => u.toString() === req.body.userId
```

Comment:
You prefer to put it in the body rather than in the url? As you want? :)

Reply:
@remirubis I forget to add the role, so now we need to put it in the body

Reply:
The other conversations can be resolved ?